text/0000-layer-origin-metadata.md

```diff
@@ -0,0 +1,234 @@
+  reproducibility puzzle; it includes the ID of the restored version
+  of the layer (if applicable), modification date, source version
+  info, and a list of the IDs of the layers that were present at layer
+  generation time.
+- **Layer ID** is the layer's unique ID, whether on disk or on the
+  registry.
+
+## Target persona
+
+This feature will be useful for **buildpack authors** (for reproducing
+layers) and **platform operators** (for reproducing builds).
+
+## Example
+
+Consider some buildpack `yabba` and some application `foo-api`. Let's
+also say that the `yabba` buildpack generates a cacheable build layer
+`yabba/build`.
+
+1. I do a build of the `foo-api` as of git SHA aaa000 involving the
+   `yabba` buildpack, without using any cache.
+2. I do a build of the `foo-api` as of git SHA aaa111 involving the
+   `yabba` buildpack, using the cache from the previous build.
+
+I want to account for all of the inputs that were present during the
+second build:
+
+1. the version of the `foo-api` application
+2. the version of the `yabba` buildpack
+3. the version of the stack's build image
+4. the version of the `yabba/build` layer that was restored from the
+   cache
+
+I can account for all but the last; I neither know that layer's
+identity, nor how it was created (the version of the `foo-api`, the
+version of the `yabba` buildpack, whatever layers existed at that
+time). Therefore, I cannot reproduce the second build.
+
+Now consider that I have access to each build's `report.toml`, and
+that the second build's `report.toml` looks something like this:
+
+```toml
+[image]
+tags = ["index.docker.io/image/name:latest", "index.docker.io/image/name:other-tag"]
+digest = "26cca5b0c787"
+previous-image-digest = "97e47fb9e0a6"
+
+[[layers]]
+buildpack = "yabba"
+name = "build"
+on-export = "reused"
+digest = "1742affe03b5"
+modification-date = "2020-07-09 12:41:35-04:00"
+size = 17813
+previous-layer-digest = "c54bba046158"
+
+cached = true
+launch = false
+build = true
+
+[[layers.application-source]]
+type = "git"
+
+[[layers.application-source.version]]
+sha = "26e8c05b151f57161d64b7a5f844bb04caed0ba3"
+```
+
+In this case, I can discover everything we need to reproduce the
+layer; if I discover later that I need to reproduced the layer
+produced by the first build, I should be able to reference the
+`report.toml` for the image designated by
+`image.previous-image-digest`.
+
+(I'm actually hand-waving around the issue of associating
+`image.previous-image-digest` and that image's `report.toml`. See
+[below][unresolved-questions].)
+
+# How it Works
+[how-it-works]: #how-it-works
+
+<!-- This is the technical portion of the RFC, where you explain the design -->
+<!-- in sufficient detail. -->
+
+<!-- The section should return to the examples given in the previous -->
+<!-- section, and explain more fully how the detailed proposal makes those -->
+<!-- examples work. -->
+
+The **builder lifecycle phase should record that it's invoking a
```

Comment:
I think it is actually the `exporter` that will need to populate this data. The exporter has access to all of the relevant information (the cache, previous image metadata, `project-metadata.toml`) and is responsible for calculating layer digests.